### PR TITLE
Fix incorrect Kubernetes feature gate defaults

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area quality
/kind bug

**What this PR does / why we need it**:
When validating Kubernetes v1.28 I noticed that several defaults of locked Kubernetes feature gates are incorrect.
This PR corrects the values for feature gates are removed in Kubernetes >=v1.25 and are locked to their defaults before. Older feature gates have not been checked because we do not support Kubernetes v1.23 and older anymore.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Please double check with the [feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) and [removed feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates-removed/) lists from Kubernetes.

**Release note**:

```bugfix operator
Several default settings of Kubernetes feature gates have been corrected.
```

### CODE REVIEW

File: pkg/utils/validation/features/featuregates.go [hunk 1]
```diff
@@ -53,21 +53,21 @@ var featureGateVersionRanges = map[string]*FeatureGateVersionRange{
...
 	"CPUManagerPolicyBetaOptions":                    {AddedInVersion: "1.23"},
 	"CPUManagerPolicyOptions":                        {},
-	"CronJobTimeZone":                                {AddedInVersion: "1.24", LockedToDefaultInVersion: "1.27"},
+	"CronJobTimeZone":                                {Default: true, AddedInVersion: "1.24", LockedToDefaultInVersion: "1.27"},
 	"CSIInlineVolume":                                {Default: true, LockedToDefaultInVersion: "1.25", RemovedInVersion: "1.27"},
 	"CSIMigration":                                   {Default: true, LockedToDefaultInVersion: "1.25", RemovedInVersion: "1.27"},
 	"CSIMigrationAWS":                                {Default: true, LockedToDefaultInVersion: "1.25", RemovedInVersion: "1.27"},
 	"CSIMigrationAzureDisk":                          {Default: true, LockedToDefaultInVersion: "1.25", RemovedInVersion: "1.27"},
-	"CSIMigrationAzureFile":                          {LockedToDefaultInVersion: "1.27"},
+	"CSIMigrationAzureFile":                          {Default: true, LockedToDefaultInVersion: "1.26"},
```
Comment (ary1992) [inline pkg/utils/validation/features/featuregates.go]:
```suggestion
	"CSIMigrationAzureFile":                          {Default: true, LockedToDefaultInVersion: "1.27"},
```

**Review by ary1992 — CHANGES_REQUESTED** [review]
Thanks.

File: pkg/utils/validation/features/featuregates.go (hunk 1 above)
Comment (oliver-goetz) [reply pkg/utils/validation/features/featuregates.go]: The report says `1.27` but according to the Kubernetes page it seems to be be `1.26`. That's weird 🤔 
```
CSIMigrationAzureFile | true | Beta | 1.24 | 1.25
CSIMigrationAzureFile | true | GA | 1.26 
```

Comment (ary1992) [reply pkg/utils/validation/features/featuregates.go]: In the release notes of 1.27 it is mentioned
> The CSIMigrationAzureFile feature gate (for the feature which graduated to GA in v1.26) is now unconditionally enabled and will be removed in v1.28. (https://github.com/kubernetes/kubernetes/pull/114953, [@enj](https://github.com/enj))

And same is true for the script output as well
```
./hack/compare-k8s-feature-gates.sh 1.26 1.27
Feature gates added in 1.27 compared to 1.26:
AdmissionWebhookMatchConditions
CloudControllerManagerWebhook
CloudDualStackNodeIPs
ClusterTrustBundle
ElasticIndexedJob
InPlacePodVerticalScaling
KubeletPodResourcesDynamicResources
KubeletPodResourcesGet
MultiCIDRServiceAllocator
NewVolumeManagerReconstruction
NodeLogQuery
SecurityContextDeny
ServiceNodePortStaticSubrange
StableLoadBalancerNodeSet
WatchList

Feature gates removed in 1.27 compared to 1.26:
CSIInlineVolume
CSIMigration
CSIMigrationAWS
CSIMigrationAzureDisk
ControllerManagerLeaderMigration
DaemonSetUpdateSurge
EphemeralContainers
ExpandCSIVolumes
ExpandInUsePersistentVolumes
ExpandPersistentVolumes
IPv6DualStack
IdentifyPodOS
LocalStorageCapacityIsolation
NetworkPolicyEndPort
StatefulSetMinReadySeconds

Feature gates locked to default in 1.27 compared to 1.26:
AdvancedAuditing
CSIMigrationAzureFile
CSIMigrationvSphere
CronJobTimeZone
DownwardAPIHugePages
GRPCContainerProbe
JobMutableNodeSchedulingDirectives
LegacyServiceAccountTokenNoAutoGeneration
OpenAPIV3
SeccompDefault
ServerSideFieldValidation
TopologyManager
```

**Comment by oliver-goetz** [comment]
It is a bit cumbersome to check the default values manually, so I enhanced the `compare-k8s-feature-gates.sh` script that it shows the default values for the locked feature gates too.

The output looks like this.
```
./hack/compare-k8s-feature-gates.sh 1.26 1.27
...
Feature gates locked to default in 1.27 compared to 1.26:
AdvancedAuditing                           Default:  true
CSIMigrationAzureFile                      Default:  true
CSIMigrationvSphere                        Default:  true
CronJobTimeZone                            Default:  true
DownwardAPIHugePages                       Default:  true
GRPCContainerProbe                         Default:  true
JobMutableNodeSchedulingDirectives         Default:  true
LegacyServiceAccountTokenNoAutoGeneration  Default:  true
OpenAPIV3                                  Default:  true
SeccompDefault                             Default:  true
ServerSideFieldValidation                  Default:  true
TopologyManager                            Default:  true
``` 

With the adapted script, I found some more defaults that have to be changed 😅

**Comment by rfranzke** [comment]
cc @ary1992

**Review by ary1992 — COMMENTED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

**Pull request merged**